osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs

```diff
@@ -343,6 +346,10 @@ private void updateBonusScore()
                 else
                     tick.TriggerResult(true);
 
+                // Find and trigger the next HP tick
+                var hpTick = ticks.FirstOrDefault(t => !t.Result.HasResult && t is DrawableSpinnerHealthTick);
+                hpTick?.TriggerResult(true);
```

Comment:
Why does this code first trigger a normal tick, and *then also* trigger a "HP" tick?

What this effectively appears to mean is that

- each "required" spin will grant a `SmallBonus` and a `HealthBonus`, for a total of 0.0085 + 0.0085 = 0.017 hp increase
- each "bonus" spin will grant a `LargeBonus` and a `HealthBonus`, for a total of 0.01 + 0.0085 = 0.0185 hp increase
- after that, each "past max" spin will only grant a `HealthBonus`, for a 0.0085 hp increase

Is there any rhyme or reason to this? Why do the normal spins gain extra hp? Why does spinning past max recover less health than spinning at any point before that?

Reply:
The logic seemed okay to me, however after changing the health ticks to only happen after the bonus, this is adjusted in a more reasonable way.